src/main/java/org/beanpod/switchboard/config/WebSecurityConfig.java

```diff
@@ -39,11 +41,16 @@ protected void configure(HttpSecurity http) throws Exception {
         .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
         .and()
         .authorizeRequests()
-        .antMatchers("/user/sign-up")
-        .permitAll()
+        .antMatchers("/user/**")
+        .hasAnyAuthority("ADMIN", "SUPERUSER")
+        .antMatchers("/**")
+        .hasAnyAuthority("USER", "SUPERUSER")
```

Comment:
What's the difference between the `hasAnyAuthority("ADMIN", "SUPERUSER")`  and `.hasAnyAuthority("USER", "SUPERUSER")` calls here?  rather, why is SuperUser in both?

Reply:
`SUPERUSER`: is a user with access to everything. It's to be used solely for development purposes.
`ADMIN`: is a user with access to only the user controller which allows them to manage/signup users.
`USER`: is a user with access to everything except the user controller.

Reply:
wait, isn't `SuperUser `actually going into the app? The PM clarified a bit ago that our Admin user would have access to the whole app + user creation

Reply:
Yes, `SUPERUSER` reflects that requirement in which we have an app user with the user creation privilege; whereas, `ADMIN` is a non-user with only user creation privilege

Reply:
Should I replace `SUPERUSER` with `ADMIN`, and remove the `SUPERUSER` role? Adding/removing roles is fairly straightforward. This can be done quickly before we merge.

Reply:
I think we can keep it like this. We have the roles we need (user and superuser). I think the admin role makes sense to have even though the PM won't use it right away, they might need it in the future.